base/irrationals.jl

```diff
@@ -136,6 +136,15 @@ hash(x::Irrational, h::UInt) = 3*objectid(x) - h
 
 widen(::Type{T}) where {T<:Irrational} = T
 
+zero(::AbstractIrrational) = false
+zero(::Type{<:AbstractIrrational}) = false
+
+one(::AbstractIrrational) = true
+one(::Type{<:AbstractIrrational}) = true
+
+oneunit(T::Type{<:AbstractIrrational}) = throw(ArgumentError("The number one cannot be of type $T"))
+oneunit(x::T) where {T <: AbstractIrrational} = oneunit(T)
```

Comment:
Why do we need this `oneunit` method, as opposed to the default `MethodError`?

Reply:
We don't but the error here is clearer. Should I remove it?

Reply:
Yes, in general people using Julia are expected to understand `MethodError`… in extremely common cases, where we need a more user-friendly message, we modify `showerror`, but I don't think that is needed here.

Reply:
Done